server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1479,7 +1479,7 @@ public static List<String> resolve(final Context context, List<String> expressio
                 } else {
                     result.add(resolveExpression(expression, context::getStartTime));
                 }
-                if (Regex.isSimpleMatchPattern(expression)) {
```

Comment:
I'm aware logic like this is not right in the `DateMathExpressionResolver`. I plan to fix this later.